contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,380 @@
+    ```
+
+3. The limit range will update the pod specification as follows:
+
+    ```yaml
+    apiVersion: v1
+    kind: pod
+    metadata:
+     name: foo
+    spec:
+     containers:
+      name: fooc
+      resources:
+       limits:
+        storage-logs: 200Mi
+        storage-overlay: 200Mi
+     volumes:
+      name: myEmptyDir
+      emptyDir:
+       capacity: 1Gi
+    ```
+
+4. Bob’s “foo” pod can use upto “200Mi” for its containers logs and writable layer each, and “1Gi” for its “myEmptyDir” volume. 
+5. If Bob’s pod “foo” exceeds the “default” storage limits and gets evicted, then Bob can set a minimum storage requirement for his containers and a higher “capacity” for his EmptyDir volumes.
+
+  ```yaml
+  apiVersion: v1
+  kind: pod
+  metadata:
+   name: foo
+  spec:
+   containers:
+    name: fooc
+    resources:
+     requests:
+      storage-logs: 500Mi
+      storage-overlay: 500Mi
+   volumes:
+    name: myEmptyDir
+    emptyDir:
+     capacity: 2Gi
+  ```
+
+6. Note that it is not possible to specify a minimum storage requirement for EmptyDir volumes because we intent to limit overcommitment of local storage due to high latency in reclaiming it from terminated pods. 
+
+### Alice manages a Database which needs access to “durable” and fast scratch space
+
+1. Cluster administrator provisions machines with local SSDs and brings up the cluster
+2. When a new node instance starts up, an addon DaemonSet discovers local “secondary” partitions which are mounted at a well known location and creates HostPath PVs for them if one doesn’t exist already. The PVs will include a path to the secondary device mount points and include additional labels and annotations that help tie to a specific node and identify the storage medium.
+
+    ```yaml
+    kind: PersistentVolume
+    apiVersion: v1
+    metadata:
+      name: local-pv-1
+      annotations:
+        storage.kubernetes.io/node: node-1
+      labels:
+        storage.kubernetes.io/medium: ssd
+    spec:
+      volume-type: local
+      storage-type: filesystem
+      capacity:
+        storage: 100Gi
+      hostpath:
+        path: /var/lib/kubelet/storage-partitions/local-pv-1
+      accessModes:
+        - ReadWriteOnce
+      persistentVolumeReclaimPolicy: Delete
+    ```
+    ```
+    $ kubectl get pv
+    NAME       CAPACITY ACCESSMODES RECLAIMPOLICY STATUS    CLAIM … NODE  
+    local-pv-1 100Gi    RWO         Delete        Available         node-1
+    local-pv-2 10Gi     RWO         Delete        Available         node-1
+    local-pv-1 100Gi    RWO         Delete        Available         node-2
+    local-pv-2 10Gi     RWO         Delete        Available         node-2
+    local-pv-1 100Gi    RWO         Delete        Available         node-3
+    local-pv-2 10Gi     RWO         Delete        Available         node-3
+    ```
+3. The addon will monitor the health of secondary partitions and taint PVs whenever the backing local storage devices becomes unhealthy.
```

Comment:
We would like to evict pods that are using tainted PVs, unbind the PVC, and reschedule the pod so that it can bind to a different PV.  I think everything after the eviction could be handled by a separate controller.